test/e2e/data/e2e_conf_cni_antrea.yaml

```diff
@@ -0,0 +1,144 @@
+    type: BootstrapProvider
+    versions:
+      - name: v0.3.14
+        # Use manifest from source files
+        value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.14/bootstrap-components.yaml"
+        type: "url"
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+          - old: --metrics-addr=127.0.0.1:8080
+            new: --metrics-addr=:8080
+
+  - name: kubeadm
+    type: ControlPlaneProvider
+    versions:
+      - name: v0.3.14
+        # Use manifest from source files
+        value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v0.3.14/control-plane-components.yaml"
+        type: "url"
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+          - old: --metrics-addr=127.0.0.1:8080
+            new: --metrics-addr=:8080
+
+  - name: aws
+    type: InfrastructureProvider
+    versions:
+      - name: v0.5.0
+        # Use manifest from source files
+        value: ../../../config
+        replacements:
+          - old: "imagePullPolicy: Always"
+            new: "imagePullPolicy: IfNotPresent"
+          - old: "--enable-leader-election"
+            new: "--enable-leader-election=false"
+          - old: --metrics-addr=127.0.0.1:8080
+            new: --metrics-addr=:8080
+          - old: gcr.io/k8s-staging-cluster-api/cluster-api-aws-controller-amd64:dev
+            new: gcr.io/k8s-staging-cluster-api/capa-manager:e2e
+          - old: gcr.io/k8s-staging-cluster-api-aws/cluster-api-aws-controller:latest
+            new: gcr.io/k8s-staging-cluster-api/capa-manager:e2e
+    files:
+      # Add a cluster template
+      - sourcePath: "./infrastructure-aws/cluster-template.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-ssm.yaml"
+        targetName: "cluster-template-ssm.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-multi-az.yaml"
+        targetName: "cluster-template-multi-az.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-limit-az.yaml"
+        targetName: "cluster-template-limit-az.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-spot-instances.yaml"
+        targetName: "cluster-template-spot-instances.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-mhc.yaml"
+        targetName: "cluster-template-mhc.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-machine-pool.yaml"
+        targetName: "cluster-template-machine-pool.yaml"
+      - sourcePath: "./infrastructure-aws/cluster-template-upgrade-to-main.yaml"
+        targetName: "cluster-template-upgrade-to-main.yaml"
+variables:
+  KUBERNETES_VERSION: "v1.20.4"
+  CNI: "../../data/cni/antrea.yaml"
```

Comment:
No need to have a separate config to use a different CNI.

CAPZ has an example of testing different CNIs with single config here:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/2d58abba334517ad59a3f2c7f6d8dd476bfb0f49/test/e2e/config/azure-dev.yaml#L85-L87

This is how they are embedding CNI resources into a template:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/4ba3dc99f2f12cd58cc3bb332b39cd9db91e1633/test/e2e/e2e_suite_test.go#L351

Reply:
Thanks @sedefsavas I'll look at this.